### PR TITLE
Create first HNS network before Flannel

### LINKED ISSUE #36: Connections to 169.254.169.254 are blocked after flannel/HNS initialization
Hitting the same issue as https://github.com/Microsoft/hcsshim/issues/299

HNS gets initialized right now in our flow when the Flannel pod runs. We need to explore options for manually re-adding a route to 169.254.169.254 ala [this code](https://github.com/GoogleCloudPlatform/compute-image-windows/blob/22bd7e2701da9dc6bd931711cbb39c36a1d81df1/sysprep/instance_setup.ps1#L126) after HNS initialization.

Options I can imagine:
- Do HNS initialization in the PrepareNode script, so we can have better control over when it happens
- Try to use a PostStart for the Flannel pod

### LINKED ISSUE #37: Flannel fails to watch subnet leases for other nodes
Related to #36 

Since Flannel is the one making the first HNS network right now, the network reconfiguration that happens is breaking the Flannel processes' ability to connect to Kubernetes.

You see in the flannel pods logs regular startup logging and then a bunch of lines like:

```
reflector.go:201] github.com/coreos/flannel/subnet/kube/kube.go:307: Failed to list *v1.Node: Get https://10.128.0.12:6443/api/v1/nodes?resourceVersion=0: http2: no cached connection was available
```

I think if we fix #36 by initializing HNS in the PrepareNode script then this might be solved too. The downside of that is that it couples the PrepareNode script to networking, we might need to add a flag to the script to specify whether we set this network up as an overlay or l2bridge network.

### PR DESCRIPTION
**Pull request by benmoss**
This is just basically [this code](https://github.com/kubernetes-sigs/sig-windows-tools/blob/ba0b4062ad1b74089b5132be506f82fcec89fdab/kubeadm/KubeClusterHelper.psm1#L607-L631), wrapped in a Go binary to make executing it through Wins easier. I tried initially to just run powershell directly with Wins, but it errored when Wins tried to rename `powershell.exe`.

This also adds back a route for 169.254.169.254, a commonly used metadata server IP. HNS network initialization breaks this on Windows.

Fixes #37
Fixes #36

/assign @gab-satchi

### CODE REVIEW

File: kubeadm/flannel/flannel-host-gw.yml [hunk 1]
```diff
@@ -29,9 +29,11 @@ data:
 
     cp -force /etc/kube-flannel/net-conf.json /host/etc/kube-flannel
     cp -force -recurse /cni/* /host/opt/cni/bin
-    cp -force /k/flannel/flanneld.exe /host/k/flannel/flanneld.exe
+    cp -force /k/flannel/* /host/k/flannel/
     cp -force /kube-proxy/kubeconfig.conf /host/k/flannel/kubeconfig.yml
     cp -force /var/run/secrets/kubernetes.io/serviceaccount/* /host/k/flannel/var/run/secrets/kubernetes.io/serviceaccount/
+    wins cli process run --path /k/flannel/setup.exe --args "--mode=overlay --interface=Ethernet"
```
Comment (gab-satchi) [inline kubeadm/flannel/flannel-host-gw.yml]: mode should be "l2bridge"

**Comment by benmoss** [comment]
@neolit123 can you lgtm this? Gab isn't a kubernetes-sigs member

**Comment by neolit123** [comment]
based on @gab-satchi 's review.
/lgtm

@gab-satchi if you have enough PRs for the kubernetes or kubernetes-sigs orgs you should request membership. with k org you are eligible for the k-sigs org.

**Comment by gab-satchi** [comment]
@neolit123 coincidentally made the membership request today 👍

**Comment by xie-qianyue** [comment]
I applied flannel-host-gw.yml, and got an error:
```
E0507 20:10:43.865945    8896 reflector.go:201] github.com/coreos/flannel/subnet/kube/kube.go:307: Failed to list *v1.Node: Get https://10.246.44.131:6443/api/v1/nodes?resourceVersion=0: http2: no cached connection was available
```
I have posted a comment [here](https://github.com/kubernetes-sigs/sig-windows-tools/issues/37#issuecomment-625219807).

Today I found a mistake in the setup.go as the review I opened above, can anybody check this review?

**Comment by benmoss** [comment]
@xie-qianyue did you find a bug? Please open a new issue rather than commenting on closed ones.

**Comment by xie-qianyue** [comment]
@benmoss OK, I open a new issue: #83.

**Pull request merged**